thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/datalayer/dto/AbstractDTO.java

```diff
@@ -90,6 +92,14 @@ public String toString() {
     return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
   }
 
+  public AccessControlIdentifiers toAccessControlIdentifiers() {
```

Comment:
Having this in the persistence layer is a bit tricky. The dtos should be pojos. Let's have an utility in the server for creating an ID from any entity.

Reply:
Ya, I wasn't really sure where to put this method. I think it makes sense to have a generic method like this and have resource/entity specific methods override it. Or maybe do a switch statement on the dto/api class.

Reply:
The conversions between dto -> api -> resource isn't super clear to me.